src/main/clojure/clara/rules/dsl.clj

```diff
@@ -47,7 +47,12 @@
   "Creates a condition with the given optional result binding when parsing a rule."
   [condition result-binding expr-meta]
   (let [type (if (symbol? (first condition))
-               (if-let [resolved (resolve (first condition))]
+               (if-let [resolved (and
+                                  ;; If we are compiling ClojureScript rules we don't want
+                                  ;; to resolve the symbol in the ClojureScript compiler's
+                                  ;; Clojure environment.  See issue 300.
+                                  (not (com/compiling-cljs?))
+                                  (resolve (first condition)))]
```

Comment:
It does seem odd to me that the clj impl would embed a class object in the data structure at this point. A fully-qualified symbol seems more appropriate and a "data representation". The fact that clj lets you embed class objects in `eval` code etc just seems to be non-documented behaviors.

This is a broader topic though, so not really suitable to be fixed in this issue.